packages/@uppy/companion/src/standalone/helper.js

```diff
@@ -9,13 +9,48 @@ const logger = require('../server/logger')
 const { version } = require('../../package.json')
 
 /**
- * Reads all companion configuration set via environment variables
- * and via the config file path
+ * Tries to read the secret from a file if the according environment variable is set.
+ * Otherwise it falls back to the standard secret environment variable.
  *
- * @returns {object}
+ * @param {string} baseEnvVar
+ *
+ * @returns {string}
  */
-exports.getCompanionOptions = (options = {}) => {
-  return merge({}, getConfigFromEnv(), getConfigFromFile(), options)
+const getSecret = (baseEnvVar) => {
+  const secretFile = process.env[`${baseEnvVar}_FILE`]
+  return secretFile
+    ? fs.readFileSync(secretFile).toString()
+    : process.env[baseEnvVar]
+}
+
+/**
+ * Auto-generates server secret
+ *
+ * @returns {string}
+ */
+exports.generateSecret = () => {
+  logger.warn('auto-generating server secret because none was specified', 'startup.secret')
+  return crypto.randomBytes(64).toString('hex')
+}
+
+/**
+ *
+ * @param {string} url
+ */
+const hasProtocol = (url) => {
+  return url.startsWith('http://') || url.startsWith('https://')
```

Comment:
nit: I think HTTPS would be more frequent than HTTP, we might as well optimize for it.
```suggestion
  return url.startsWith('https://') || url.startsWith('http://')
```